src/adera/Machines/RCSController.cpp

```diff
@@ -0,0 +1,125 @@
+#include "RCSController.h"
+#include "osp/Active/ActiveScene.h"
+
+#include <Magnum/Math/Vector.h>
+
+using namespace adera::active::machines;
+using namespace osp::active;
+using Magnum::Vector3;
+
+
+MachineRCSController::MachineRCSController()
+    : Machine(true),
+    m_wiCommandOrient(this, "Orient"),
+    m_woThrottle(this, "Throttle")
+{
+    using wiretype::Percent;
+    m_woThrottle.value() = Percent{0.0f};
+}
+
+void MachineRCSController::propagate_output(WireOutput* output)
+{
+
+}
+
+WireInput* MachineRCSController::request_input(osp::WireInPort port)
+{
+    return existing_inputs()[port];
+}
+
+WireOutput* MachineRCSController::request_output(osp::WireOutPort port)
+{
+    return existing_outputs()[port];
+}
+
+std::vector<WireInput*> MachineRCSController::existing_inputs()
+{
+    return {&m_wiCommandOrient};
+}
+
+std::vector<WireOutput*> MachineRCSController::existing_outputs()
+{
+    return {&m_woThrottle};
+}
+
+SysMachineRCSController::SysMachineRCSController(ActiveScene& scene)
+    :SysMachine<SysMachineRCSController, MachineRCSController>(scene),
+    m_updateControls(scene.get_update_order(), "mach_rcs", "wire", "controls",
+        [this](ActiveScene& rScene) { this->update_controls(rScene); })
+{
+}
+
+float SysMachineRCSController::thruster_influence(Vector3 posOset, Vector3 direction,
+    Vector3 cmdTransl, Vector3 cmdRot)
+{
+    using Magnum::Math::cross;
+    using Magnum::Math::dot;
+    using Magnum::Math::clamp;
+
+    Vector3 thrust = -direction.normalized();
+    
+    float rotInfluence = 0.0f;
+    float translInfluence = 0.0f;
+
+    if (cmdRot.length() > 0.0f)
+    {
+        Vector3 torque = cross(posOset, thrust).normalized();
+        rotInfluence = dot(torque, cmdRot.normalized());
+    }
+
+    if (cmdTransl.length() > 0.0f)
+    {
+        translInfluence = dot(thrust, cmdTransl.normalized());
+    }
+
+    float total = rotInfluence + translInfluence;
+
+    if (total < .01f)
+    {
+        // Ignore small contributions from imprecision
+        // Real thrusters can't throttle this deep anyways
+        return 0.0f;
+    }
+    return clamp(rotInfluence + translInfluence, 0.0f, 1.0f);
```

Comment:
Is this any different than std::clamp? If not, just use std::clamp

Reply:
Nope, it's just Magnum's built-in version. I can switch it though.